app/assets/javascripts/nested_field.js.coffee

```diff
@@ -54,11 +54,59 @@
         id = getLastChildId(last_child)
 
         if id != -1
-          html = last_child.html()
           time = new Date().getTime() # use current time to make sure the new id is unique
-          html = html.replace(new RegExp("\\["+id+"\\]", 'g'),'['+(id+time)+']')
-          html = html.replace(new RegExp('_'+id+'_', 'g'), '_'+(id+time)+'_')
-          last_child.html(html)
+          last_child.find('input').each (i, e) =>
+            name = $(e).prop('name').replace(/\[\d+\]/g, "[#{time}]")
+            $(e).prop('name', name)
+            id = $(e).prop('id').replace(/_\d+_/g, "_#{time}_")
+            $(e).prop('id', id)
+            return
       , 15)
     )
+
+    resetNestedItemId = (item) ->
+      # remove id
+      item.find("input[id$='_id']").remove()
+
+      # get child id
+      id = getLastChildId(item)
+
+      if id != -1
+        html = item.html()
+        time = new Date().getTime() # use current time to make sure the new id is unique
+        html = html.replace(new RegExp("\\["+id+"\\]", 'g'),'['+(id+time)+']')
```

Comment:
@straleyb I believe these RegEx still need to be replaced, do you recall our work above on lines 60ish?

Reply:
Yeah, about changing the props instead of replacing the whole html. I can translate that work down into this if block as well.